### PR TITLE
Generator twig

### PR DESCRIPTION
**Pull request by rcaracaus**
Steps to test:

You can use /demo-twig, which I pulled out of this repo in [last commit](https://github.com/rcaracaus/kss-node/commit/a61285d58e01a5f5b5d6089c34368b9a85a8c2e7) as a sample source to test this PR.

run:
bin/kss-node --config demo-twig/kss-config.json --verbose

### CODE REVIEW

File: generator/twig/template/public/prettify.js [hunk 1]
```diff
@@ -0,0 +1,1477 @@
+// Copyright (C) 2006 Google Inc.
```
Comment (rcaracaus) [inline generator/twig/template/public/prettify.js]: There is a lot of duplication here of front-end assets, maybe there is a way we can abstract?

File: generator/twig/template/index.html [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    <ul class="kss-nav__menu">
+      <li class="kss-nav__menu-item">
+        <a href="./">
+          <span class="kss-nav__ref">0</span>
+          <span class="kss-nav__name">Overview</span>
+        </a>
+      </li>
+      {% for section in styleguide.data.sections %}
+        {% if section.depth == 1 %}
+          <li class="kss-nav__menu-item">
```
Comment (rcaracaus) [inline generator/twig/template/index.html]: I feel like this could be turned into a twig block or something, or a kss-node provided component include

**Comment by rcaracaus** [comment]
It looks like https://github.com/justjohn/twig.js just bumped to 0.8.4, this PR is 0.8.2

could probably bump this PR to use that

**Comment by rcaracaus** [comment]
@JohnAlbin Could you have  a quick scan of this when you get a chance, happy to fix any preliminary issues.

**Comment by queenvictoria** [comment]
To test
```bash
$ git pull https://github.com/rcaracaus/kss-node.git kss
$ cd kss
$ git checkout -b generator-twig origin/generator-twig
$ npm install
```
and in my case
```
$ cd ../../
$ grunt styleguide
```

First issue I have is my customisation of the index.json file. I use a *base* parameter to alter the route to the template files. This is supported in twig but not in kss.
```JSON
{
  "template": "../../src/styleguide",
  "source": "../../src/less",
  "base": "../../src/twig",
  "destination": "../../dist/styleguide",
  "css": [
  	"../../dist/libs/bootstrap/css/bootstrap.min.css", 
  	"../../dist/css/font-awesome.css", 
  	"../../dist/css/style-1.css",
  	"../../dist/css/style-2.css"
  ],
  "js": ["../../dist/libs/modernizr/modernizr.js"]
}
```

Second issue is that the twig implementation doesn't support filters. In my implementation I have things like
```twig
  <link rel="stylesheet" href="public/kss.css">
  {{ styles|raw }}
</head>
```
and
```twig
{{ section.markup|render|escape }}<
```

I end up with styleguides that have escaped HTML and unparsed twig. Maybe we should do a beer and a sprint in the same room?

**Comment by rcaracaus** [comment]
@queenvictoria 

I am registering the twig templates in the [kss_twig_generator.js](https://github.com/rcaracaus/kss-node/blob/c11c05a945ec58b985ca8671d0247548ee5f502f/generator/twig/kss_twig_generator.js#L187) rather than through the render filter. This way I am able to do includes directly in the index.html in the template. 

Default twig filters like escape are working fine for me, and I was able to get custom extensions working as well. It doesn't seem like we need to use custom twig extensions to load templates.. unless there was a special usecase for this.

I do not really understand why you need a base directory.

{{ styles | raw }} seems to work for me, but it also seemed to work without the filter.. probably missing something simple here.

**Comment by rcaracaus** [comment]
just realised this needs support for data-kss="lorem-ipsum" too ..

File: bin/kss-node [hunk 3]
```diff
@@ -292,11 +292,14 @@ yargs
...
 
 // And re-parse the options.
+
 if (generatorConfig.xdemo) {
   generatorConfig = yargs.parse(process.argv.slice(2));
 } else {
   generatorConfig = yargs.argv;
 }
+
+
```
Comment (RobLoach) [inline bin/kss-node]: Whitespace changes?

**Comment by immoreel** [comment]
Is there still any work being done on the twig integration?
It would be really awesome to have...

**Comment by rcaracaus** [comment]
This would be good to have but first this branch needs to be rerolled on top of JohnAlbin's latest changes in master. Then we need to have some way to make sure that my generator doesn't cause any regressions.

**Comment by JohnAlbin** [comment]
#272 is fixed.

**Pull request closed without merging**